Source/options.cpp

```diff
@@ -839,56 +848,82 @@ void OptionEntryLanguageCode::LoadFromIni(string_view category)
-		if (utf8Buffer[2] == '-') {
-			// if a region is included in the locale do a simple transformation to the expected POSIX style.
-			utf8Buffer[2] = '_';
-		}
+		// if a region is included in the locale do a simple transformation to the expected POSIX style.
+		std::replace(utf8Buffer, utf8Buffer + sizeof(utf8Buffer), '-', '_');
+		return std::move(std::string(utf8Buffer));
+	};
+	WCHAR localeBuffer[LOCALE_NAME_MAX_LENGTH];
 
-		locale.append(utf8Buffer);
+	if (GetUserDefaultLocaleName(localeBuffer, LOCALE_NAME_MAX_LENGTH) != 0) {
+		// Found a user default locale convert from WIN32's default of UTF16 to UTF8 and add to the list
+		locales.push_back(wideCharToUtf8(localeBuffer));
+	}
+
+	ULONG numberOfLanguages;
+	ULONG bufferSize = LOCALE_NAME_MAX_LENGTH;
+	GetUserPreferredUILanguages(MUI_LANGUAGE_NAME, &numberOfLanguages, localeBuffer, &bufferSize);
+
+	PWSTR bufferOffset = localeBuffer;
+	for (unsigned i = 0; i < numberOfLanguages; i++) {
+		// Found one (or more) user preferred UI language(s), add these to the list to check as well
+		locales.push_back(wideCharToUtf8(bufferOffset));
+
+		// GetUserPreferredUILanguages returns a null separated list of strings, need to increment past the null terminating the current string.
+		bufferOffset += lstrlenW(localeBuffer) + 1;
 	}
 #else
 	// Fallback method for older versions of windows, this is deprecated since Vista
 	char localeBuffer[LOCALE_NAME_MAX_LENGTH];
 	// Deliberately not using the unicode versions here as the information retrieved should be represented in ASCII/single byte UTF8 codepoints.
 	if (GetLocaleInfoA(LOCALE_USER_DEFAULT, LOCALE_SISO639LANGNAME, localeBuffer, LOCALE_NAME_MAX_LENGTH) != 0) {
-		locale.append(localeBuffer);
+		std::string locale { localeBuffer };
 		if (GetLocaleInfoA(LOCALE_USER_DEFAULT, LOCALE_SISO3166CTRYNAME, localeBuffer, LOCALE_NAME_MAX_LENGTH) != 0) {
 			locale.append("_");
 			locale.append(localeBuffer);
 		}
+		locales.push_back(std::move(locale));
 	}
 #endif
 #else
-	std::string locale = std::locale("").name().substr(0, 5);
+	locales.emplace_back(std::locale("").name().substr(0, 5));
 #endif
 
-	locale = locale.substr(0, 5);
-	LogVerbose("Preferred locale: {}", locale);
-	if (!HasTranslation(locale)) {
-		locale = locale.substr(0, 2);
-		if (!HasTranslation(locale)) {
-			locale = "en";
+	// Insert non-regional locale codes after the last regional variation so we fallback to neutral translations if no regional translation exists that meets user preferences.
+	for (auto localeIter = locales.rbegin(); localeIter != locales.rend(); localeIter++) {
+		auto regionSeparator = localeIter->find('_');
+		if (regionSeparator != std::string::npos) {
+			std::string neutralLocale = localeIter->substr(0, regionSeparator);
+			if (std::find(locales.rbegin(), localeIter, neutralLocale) == localeIter) {
+				localeIter = std::make_reverse_iterator(locales.insert(localeIter.base(), neutralLocale));
+			}

```

Comment:
This assumes that user locales will always be regional variations, but if a user does have a non-regional locale earlier in the list then at worst we're duplicating the check one extra time.